src/mlpack/bindings/R/mlpack/tests/testthat/test-R_binding.R

```diff
@@ -0,0 +1,297 @@
+      expect_true(output$matrix_and_info_out[j, i] == 2 * x[j, i])
+    }
+  }
+
+  for (j in 1:10) {
+    expect_true(output$matrix_and_info_out[j, 10] == j)
+  }
+})
+
+# Test that we can pass a data.frame with numeric and categorical features.
+test_that("TestDataFrameWithLogicalInfo", {
+  y <- matrix(rexp(90, rate = .1), ncol = 9)
+  x <- data.frame(y)
+  x["e"] <- c(T, F, F, T, T, F, F, F, F, T)
+
+  output <- test_r_binding(4.0, 12, "hello",
+                           matrix_and_info_in=x)
+
+  expect_identical(dim(output$matrix_and_info_out), as.integer(c(10, 10)))
+
+  for (i in 1:9) {
+    for (j in 1:10) {
+      expect_true(output$matrix_and_info_out[j, i] == 2 * x[j, i])
+    }
+  }
+  expect_identical(output$matrix_and_info_out[, 10], as.numeric(x[, "e"]))
+})
+
+# Test that we can pass a vector of ints and get back that same vector but with
+# the last element removed.
+test_that("TestIntVector", {
+  x <- c(1, 2, 3, 4, 5)
+
+  output <- test_r_binding(4.0, 12, "hello",
+                           vector_in=x)
+
+  expect_identical(output$vector_out, c(1:4))
+})
+
+# Test that we can pass a vector of strings and get back that same vector but
+# with the last element removed.
+test_that("TestStringVector", {
+  x <- letters[1:5]
+
+  output <- test_r_binding(4.0, 12, "hello",
+                           str_vector_in=x)
+
+  expect_identical(output$str_vector_out, letters[1:4])
+})
+
+# If we give data other than matrix/data.frame in matrix_in/matrix_and_info_in,
+# we should get an error. 
+test_that("TestNotMatrix", {
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_in="wrong"))
+
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_in=12))
+
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_in=1e6))
+
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_and_info_in="wrong"))
+
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_and_info_in=12))
+
+  expect_error(test_r_binding(4.0, 12, "hello",
+                              matrix_and_info_in=1e6))
+})
+
+# First create a GaussianKernel object, then send it back and make sure we get
+# the right double value.
+test_that("TestModel", {
+  output1 <- test_r_binding(4.0, 12, "hello",
+                            build_model=TRUE)
+
+  output2 <- test_r_binding(4.0, 12, "hello",
+                            model_in=output1$model_out)
+
+  expect_true(output2$model_bw_out == 20)
+})
+
+# Test that we can serialize a model to disk and then use it again.
+test_that("TestSerialization", {
+  output1 <- test_r_binding(4.0, 12, "hello",
+                            build_model=TRUE)
+
+  serialize_gaussian_kernel("model.bin", output1$model_out)
+
+  new_model <- unserialize_gaussian_kernel("model.bin")
+  unlink("model.bin")
+
+  output2 <- test_r_binding(4.0, 12, "hello", model_in = new_model)
```

Comment:
```suggestion
  output2 <- test_r_binding(4.0, 12, "hello", model_in=new_model)
```
Just to match the other bits. :)

Reply:
Done :)